### PR TITLE
dont pass an explicit --native-null-assertion option if none was provided

### PR DESCRIPTION
**Pull request by jakemac53**
Related to https://github.com/dart-lang/sdk/issues/50710, this will allow the optimization level in dart2js to control the default setting for this flag. It can still be overridden with explicit config.

### CODE REVIEW

File: build_web_compilers/lib/src/dev_compiler_bootstrap.dart [hunk 1]
```diff
@@ -193,12 +193,15 @@ String _appBootstrap({
   required String moduleScope,
   required String entrypointLibraryName,
   required String oldModuleScope,
-  required bool nativeNullAssertions,
-}) =>
-    '''
+  required bool? nativeNullAssertions,
+}) {
+  var nativeAssertsCode = nativeNullAssertions == null
```
Comment (jakemac53) [inline build_web_compilers/lib/src/dev_compiler_bootstrap.dart:198]: cc @nshahan for this part in particular, I presume it is fine to simply omit this line?

Comment (srujzs) [reply build_web_compilers/lib/src/dev_compiler_bootstrap.dart:198]: It looks like we default to false in DDC (we still check in sound mode though) if we never call the `nativeNonNullAsserts` function, which is consistent with dart2js, so this looks good.

**Review by srujzs — APPROVED** [review]

File: build_web_compilers/lib/src/dart2js_bootstrap.dart [hunk 2]
```diff
@@ -24,15 +24,15 @@ import 'web_entrypoint_builder.dart';
 Future<void> bootstrapDart2Js(
   BuildStep buildStep,
   List<String> dart2JsArgs, {
-  required bool nativeNullAssertions,
+  required bool? nativeNullAssertions,
```
Comment (natebosch) [inline build_web_compilers/lib/src/dart2js_bootstrap.dart:27]: [optional] Doesn't need to be required anymore

**Review by natebosch — APPROVED** [review]

File: build_web_compilers/lib/src/dart2js_bootstrap.dart (hunk 2 above)
Comment (jakemac53) [reply build_web_compilers/lib/src/dart2js_bootstrap.dart:27]: I like making it required to be explicit even though it's nullable

File: build_web_compilers/lib/src/dev_compiler_bootstrap.dart (hunk 1 above)
Comment (jakemac53) [reply build_web_compilers/lib/src/dev_compiler_bootstrap.dart:198]: Ok, that does sound like a behavior change (we defaulted it on before), but regardless it seems best to leave this up to the compilers.

**Pull request merged**